### DISCUSSION TITLE
VSCode terminal copilot append path with space in $PATH

### DISCUSSION THREAD

**Discussion by Row** [community/community#147401] [Copilot Conversations]
### Select Topic Area

Question

### Body

Howdy.

In my setup/project, which use makefiles, the build fails because copilot appends a path with a space to the $PATH env in VSCode.

In the lines of:
```
....:/Users/username/Library/Application Support/Code/User/globalStorage/github.copilot-chat/debugCommand
```

When running make it complains that the
```
env: Support/Code/User/globalStorage/github.copilot-chat/debugCommand: No such file or directory
```

This could probably be solved on a project level, but I rather just disable the feature in Copilot.

Is it possible?
I tried to disable the terminal in the copilot settings, it did not help.

E.g. if I manually reset the $PATH to not include the changes it works fine.

Thanks.

**Comment by Misfits09** [comment]
I'm facing the same issue, disabled copilot for a while waiting for it to be patched but no one seems to care 😬 

I ended up adding this to my bashrc/zshrc file to espace the space
`export PATH=$(echo "$PATH" | sed 's/ /\\ /g')`

**Comment by michaely-cb** [comment]
Also facing the same issue. Apparently this did not happen before today for me, so maybe some auto update triggered this behaviour. My usage was from `make` where the `PATH` was evaluated in the process. I have to bring up another terminal that's not in VSCode and `make` from there.

**Comment by dgpokl** [comment]
For me, this bug also produces these 4 ugly lines in every new vsc terminal. ~~I suppose it's probably the result of something like chruby or other initialization scripts.~~
```
(eval):1: no such file or directory: Support/Code/User/globalStorage/github.copilot-chat/debugCommand
(eval):1: no such file or directory: Support/Code/User/globalStorage/github.copilot-chat/debugCommand
(eval):1: no such file or directory: Support/Code/User/globalStorage/github.copilot-chat/debugCommand
(eval):1: no such file or directory: Support/Code/User/globalStorage/github.copilot-chat/debugCommand
```
**UPDATE:** While trying to add a similar fix as Misfits09 pointed out at the top of my .zshrc, I've determined that the errors are actually emitted before my .zshrc even runs, so I can't fix it that way. So I guess VSC is actually running its command(s) that mess up the PATH first, then whatever 4 things don't like the path, and then execing your shell.

I was able to fix this though by turning off the setting Github - Copilot - Chat - Copilot Debug Command, aka:
"github.copilot.chat.copilotDebugCommand.enabled": false

**Reply by jordanliu** [reply]
ty <3
